### DISCUSSION TITLE
I want to import the notebook Plot file into a Pdf file.

### DISCUSSION THREAD

**Discussion by talhash02** [AammarTufail/python_ka_chilla2023#61] [Q&A]
Hello member,
I want to import the notebook Plot file into a Pdf file. but I am facing an error.

**If you have not installed xelatex (TeX) you will need to do so before you can export to PDF, for further instructions please look https://nbconvert.readthedocs.io/en/latest/install.html#installing-tex. 
To avoid installing xelatex (TeX) you might want to try exporting to HTML and using your browser's "Print to PDF" feature.**
I am much tried to install latex library that supports these files, but not been resolved.
Please guide me about this.
Thanks.

**Comment by Muhammad-Ali-Butt** [comment]
check this discussion out and tell me if it worked or not :

https://github.com/AammarTufail/python_ka_chilla2023/discussions/29


if not then i can explain another way to directly export ipynb as pdf but that way is a bit tedious and requires quite some time.

**Reply by talhash02** [reply]
you have to install MIKTIX from below mentioned link

https://miktex.org/download

instal pandoc

https://pandic.org/installing.html

You can convert an IPython Notebook file (with the .ipynb extension) to a PDF file using the following steps in Visual Studio Code:

Install the nbconvert package using pip:
Copy code
pip install nbconvert
pip install pyppeteer

Open the terminal in Visual Studio Code by selecting Terminal > New Terminal from the top menu.

Change the current working directory to the location of your .ipynb file.

## **After installation of all software its shows same error**

Run the following command:

Copy code
jupyter nbconvert --to pdf filename.ipynb

Replace "filename" with the actual name of your .ipynb file. This will generate a PDF file with the same name as the .ipynb file.

**Comment by Muhammad-Ali-Butt** [comment]
I will tell you the method through which i managed to export the ipynb file as pdf directly but it would take time 
uninstall miktex if you by any chance updated it after installation or opened it and start from scratch 
also i would suggest you do this in morning because it will a lot of time. 

If you have installed miktex and havent updated or messed with it then install pandoc from chrome and move to Step 3) , 4) and 5) mentioned below in steps for first timers and follow them.



**Steps For first timers, follow these steps below:** or Those starting from scratch again to avoid complications:

1) first install miktex and after installation you might get some notification to update miktex files etc **Do not click on it** just simply install miktex and do not open it.

2) then install pandoc 

3) after this open vs code and install TeX , nbconvert ,  by the following command: 
pip install TeX
pip install nbconvert
close vscode and open again and open your ipynb file.

4) Set up local repository by moving to 7:13 timestamp of the video: https://www.youtube.com/watch?v=EITY3U4lHus  and make sure you have at least 5gb of free space in local disk C. Also keep an eye once you run the command for local repository as specified in the video because the download can stop depending upon your internet connection and you would need to check that and press something and it will start again from where it stopped.

5) After the local repository has been set up , go to vscode where your ipynb file is opened and click on the 3 dots next to outline and select export and then choose pdf and then just carefully follow the video from timestamp 5:02 to 7:10 and then follow the rest of the video from timestamp 9:09 to 12:08.
